custom/icds_reports/reports/ls_launched.py

```diff
@@ -0,0 +1,315 @@
+            districts=Sum('num_launched_districts') if level <= 2 else Max('num_launched_districts'),
+            blocks=Sum('num_launched_blocks') if level <= 3 else Max('num_launched_blocks'),
+            sectors=Sum('num_launched_supervisors') if level <= 4 else Max('num_launched_supervisors'),
+            awcs=Sum('num_launched_awcs') if level <= 5 else Max('num_launched_awcs'),
+            ls_launched=Sum('num_supervisor_launched'),
+        ).order_by('%s_name' % loc_level, '%s_map_location_name' % loc_level)
+
+        if not show_test:
+            queryset = apply_exclude(domain, queryset)
+        return queryset
+
+    data_for_map = defaultdict(lambda: {
+        'awcs': [],
+        'sectors': [],
+        'blocks': [],
+        'districts': [],
+        'states': [],
+        'original_name': [],
+        'ls_launched': []
+    })
+
+    for row in get_data_for(config):
+        name = row['%s_name' % loc_level]
+        on_map_name = row['%s_map_location_name' % loc_level] or name
+        awcs = row['awcs'] or 0
+        sectors = row['sectors'] or 0
+        blocks = row['blocks'] or 0
+        districts = row['districts'] or 0
+        states = row['states'] or 0
+        ls_launched = row['ls_launched'] or 0
+
+        data_for_map[on_map_name]['awcs'].append(awcs)
+        data_for_map[on_map_name]['sectors'].append(sectors)
+        data_for_map[on_map_name]['blocks'].append(blocks)
+        data_for_map[on_map_name]['districts'].append(districts)
+        data_for_map[on_map_name]['states'].append(states)
+        data_for_map[on_map_name]['original_name'].append(name)
+        data_for_map[on_map_name]['ls_launched'].append(ls_launched)
+
+    for data_for_location in data_for_map.values():
+        data_for_location['ls_launched'] = (
+            sum(data_for_location['ls_launched'])
+        )
+        data_for_location['awcs'] = (
+            sum(data_for_location['awcs']) if level <= 5 else max(data_for_location['awcs'])
+        )
+        data_for_location['sectors'] = (
+            sum(data_for_location['sectors']) if level <= 4 else max(data_for_location['sectors'])
+        )
+        data_for_location['blocks'] = (
+            sum(data_for_location['blocks']) if level <= 3 else max(data_for_location['blocks'])
+        )
+        data_for_location['districts'] = (
+            sum(data_for_location['districts']) if level <= 2 else max(data_for_location['districts'])
+        )
+        data_for_location['states'] = (
+            sum(data_for_location['states']) if level <= 1 else max(data_for_location['states'])
+        )
+        data_for_location.update({'fillKey': 'Launched' if data_for_location['ls_launched'] > 0 else 'Not launched'})
+
+    if level == 1:
+        prop = 'states'
+    elif level == 2:
+        prop = 'districts'
+    elif level == 3:
+        prop = 'blocks'
+    elif level == 4:
+        prop = 'sectors'
+    else:
+        prop = 'awcs'
```

Comment:
looks like this is duplicated below and could be pulled into a function (assuming one doesn't exist already)